### PR TITLE
doc: add Plugin version to store and verifier crd

### PR DESCRIPTION
**Pull request by susanshi**

### CODE REVIEW

File: docs/reference/crds/verifiers.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ spec:
   name: required, name of the verifier
   artifactType: required, the type of artifact this verifier handles
   address: optional. Plugin path, defaults to value of env "RATIFY_CONFIG" or "~/.ratify/plugins"
+  version: optional. Version of the plugin implementation, defaults to 1.0.0
```
Comment (FeynmanZhou) [inline docs/reference/crds/verifiers.md:12]: Is there any other values that users can choose to fill it out in this field apart from 1.0.0?

Comment (susanshi) [reply docs/reference/crds/verifiers.md:12]: Yes, the verion depends on the range of version the verifier supports. For example, the SBOM verifier is at 2.0.0 alpha

Comment (FeynmanZhou) [reply docs/reference/crds/verifiers.md]: Adding an index table to list all available versions of external plugins would be much more friendly. Pointing to plugin code is not that friendly.

Comment (binbin-li) [reply docs/reference/crds/verifiers.md]: I feel the supported versions of each verifier should be listed in the verifier repo and Ratify-web has link to it. However, currently we don't have it in the Ratify repo yet.

Comment (susanshi) [reply docs/reference/crds/verifiers.md]: As discussed during community, i ve added the index page to include info from plugin code. Please review when convenient thanks!

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Review by binbin-li — APPROVED** [review]

**Pull request merged**